accepted/file-enumeration.md

```diff
@@ -0,0 +1,238 @@
+# Extensible File Enumeration
+
+**PM** [Immo Landwerth](https://github.com/terrajobst) | **Dev** [Jeremy Kuhne](https://github.com/jeremykuhne)
+
+Enumerating files in .NET provides limited configurability. You can specify a simple DOS style pattern and whether or not to look recursively. More complicated filtering requires post filtering all results which can introduce a significant performance drain.
+
+Recursive enumeration is also problematic in that there is no way to handle error states such as access issues or cycles created by links.
+
+These restrictions have a significant impact on file system intensive applications, a key example being MSBuild. We can address these restrictions and provide performant, configurable enumeration.
```

Comment:
>We can address these restrictions and provide performant, configurable enumeration. [](start = 113, length = 83)

You want to make sure that your intro has a concrete proposal. Maybe something like:

> This proposes a new set of primitive file and directory traversal APIs that are optimized for minimal overhead in order to make enumeration both more flexible as well as more performant.